### PR TITLE
Use likelihood instead of probability in `ndt`

### PR DESCRIPTION
**Pull request by grischi**
The score function of NDT as implemented based on [Magnusson 2009], Eq. 6.9 & 6.10 is a likelihood, not a probability. This may sound like an irrelevant naming detail, but may cause confusion for example when obtaining values larger than 1 from getTransformationProbability. This PR replaces "probability" with "likelihood".

This is the same like PR #5011, which i will close. Sorry for the inconvenience. I had to do this because i had to rename the source branch.

### CODE REVIEW

File: registration/include/pcl/registration/ndt.h [hunk 1]
```diff
@@ -483,9 +496,16 @@ class NormalDistributionsTransform : public Registration<PointSource, PointTarge
...
   double gauss_d1_, gauss_d2_;
 
-  /** \brief The probability score of the transform applied to the input cloud,
+  /** \brief The likelihood score of the transform applied to the input cloud,
    * Equation 6.9 and 6.10 [Magnusson 2009]. */
-  double trans_probability_;
+  union {
+    PCL_DEPRECATED(1,
+                   16,
+                   "The member `trans_probability_` is deprecated. Please use "
```
Comment (kunaltyagi) [inline registration/include/pcl/registration/ndt.h:504]: Note: "is deprecated" is already added by the macro.

File: registration/include/pcl/registration/ndt.h [hunk 2]
```diff
@@ -168,13 +168,26 @@ class NormalDistributionsTransform : public Registration<PointSource, PointTarge
...
+  {
+    return trans_likelihood_;
+  }
+
   /** \brief Get the registration alignment probability.
    * \return transformation probability
    */
+  PCL_DEPRECATED(1,
+                 16,
+                 "The method `getTransformationProbability` was renamed to "
```
Comment (kunaltyagi) [inline registration/include/pcl/registration/ndt.h:185]:
```suggestion
                 "The method `getTransformationProbability` has been renamed to "
```

**Review by kunaltyagi — CHANGES_REQUESTED** [review]

File: registration/include/pcl/registration/ndt.h (hunk 1 above)
Comment (grischi) [reply registration/include/pcl/registration/ndt.h]: done

File: registration/include/pcl/registration/ndt.h (hunk 2 above)
Comment (grischi) [reply registration/include/pcl/registration/ndt.h]: done

**Review by kunaltyagi — APPROVED** [review]

**Review by larshg — APPROVED** [review]

**Comment by grischi** [comment]
In addition I would like to mention that the score function actually only is the log likelihood if the scale factors c_0 are computed correctly and not approximated by a constant, see #5074.  But no matter how, it is always a log likelihood rather than a probability.

**Review by mvieth — APPROVED** [review]

**Pull request merged**